### PR TITLE
use hierarchy to break the chains of nodes in consolidation

### PR DESCRIPTION
**Pull request by martinfleis**

### CODE REVIEW

File: sgeop/nodes.py [hunk 1]
```diff
@@ -402,9 +402,12 @@ def consolidate_nodes(
 
     Replace clusters of nodes with a single node (weighted centroid
     of a cluster) and snap linestring geometry to it. Cluster is
-    defined using DBSCAN on coordinates with ``tolerance==eps`.
+    defined using hierarchical clustering with average linkage
+    on coordinates cut at a cophenetic distance equal to ``tolerance`.
```
Comment (jGaboardi) [inline sgeop/nodes.py:406]:
```suggestion
    on coordinates cut at a cophenetic distance equal to ``tolerance``.
```

**Review by jGaboardi — APPROVED** [review]
Before we fix tests here should we work out #132 ?

**Comment by jGaboardi** [comment]
Are we comfortable with the [fresh results](https://github.com/uscuni/sgeop/actions/runs/12141194139#artifacts) being "the new truth?" Is so, I can add them to this PR to get the testing green and we can merge.

**Comment by martinfleis** [comment]
I didn't even notice that some tests are failing on CI, all worked locally :D. I'd update tests here and then in the other one. Add the new stuff please.

**Review by jGaboardi — APPROVED** [review]
Passing CI with fresh truth. I'll merge once all OS runs are green.

**Pull request merged**